scripts/run-cmake-format.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# This script is a wrapper for cmakelang that may be used with pre-commit. The
+# wrapping is necessary because RAPIDS libraries split configuration for
+# cmakelang linters between a local config file and a second config file that's
+# shared across all of RAPIDS via rapids-cmake. In order to keep it up to date
+# this file is only maintained in one place (the rapids-cmake repo) and
+# pulled down during builds. We need a way to invoke CMake linting commands
+# without causing pre-commit failures (which could block local commits or CI),
+# while also being sufficiently flexible to allow users to maintain the config
+# file independently of a build directory.
+#
+# This script provides the minimal functionality to enable those use cases. It
+# searches in a number of predefined locations for the rapids-cmake config file
+# and exits gracefully if the file is not found. If a user wishes to specify a
+# config file at a nonstandard location, they may do so by setting the
+# environment variable RAPIDS_CMAKE_FORMAT_FILE.
+# 
+# This script can be invoked directly anywhere within the project repository.
+# Alternatively, it may be invoked as a pre-commit hook via
+# `pre-commit run (cmake-format)|(cmake-lint)`.
+#
+# Usage:
+# bash run-cmake-format.sh {cmake-format,cmake-lint} infile [infile ...]
+
+status=0
+if [ -z ${RMM_ROOT:+PLACEHOLDER} ]; then
+    RMM_ROOT=$(git rev-parse --show-toplevel 2>&1)/build
```

Comment:
I think we want to refactor `RMM_ROOT` to be named `RMM_BUILD_DIR` since we are currently storing
 <git_root>/build in the variable. 
 
 This will also allow us to clarify the error messages on 36 where we can say "error attempting to determine project build directory"

Reply:
I agree that this naming seems a bit inconsistent with what we're used to. I made this choice to match rapids-compose, which sets `RMM_ROOT=${RMM_HOME}/build/release`. So $RMM_HOME is used for the root of the repo, while RMM_ROOT is the root of the build. What do you think would be the most consistent choice?

Reply:
Matching existing naming conventions is reasonable. I think expanding the error message to be something like the following would remove ambiguity about source vs build dir.
```
"This script must be run inside the rmm repository, or the RMM_ROOT environment variable must be set to the RMM build directory."
```

Reply:
What's wrong with RMM_BUILD_DIR? I don't think rapids-compose should be the source of truth here.

Reply:
I agree, I don't think compose needs to be the arbiter of truth in this scenario. My question was basically aimed at getting at how this should behave inside a compose container given that it sets RMM_ROOT to the build directory (specifically, since compose supports separate builds for separate branches it uses the symlinked build directory for a specific build type, e.g. release/relwithdebinfo/etc). In a normal context (not rapids-compose), is that where we expect a `<PackageName>_ROOT` variable to point?

Reply:
I don't know either.

Reply:
For CMake `<PackageName>_ROOT` points to the root of the install tree ( `/usr/local/`, `<my/conda/env>` ) or can point to the build directory of a project that hasn't been installed.

I don't expect all developers understand CMake variable naming convention and so using an explicitly clear name `RMM_BUILD_DIR` would make everything less ambiguous

Reply:
Switched to RMM_BUILD_DIR.